File: LettercasePercentageRatio/lcr.rb
# lettercase percentage ratio soln in rb for codeeval by steven a dunn

File.open(ARGV[0], 'r').each_line do |line|
    line.chomp!

    upper_count = 0.0
    lower_count = 0.0
    chars = line.split("").map do |x|
        if x =~ /[A-Z]/
            upper_count += 1
        else
            lower_count += 1
        end
    end

    total_count = upper_count + lower_count
    upper_percent = (upper_count / total_count) * 100
    lower_percent = (lower_count / total_count) * 100

    printf("lowercase: %.2f uppercase %.2f \n", lower_percent, upper_percent)
end

File: LettercasePercentageRatio/c/lcr.c
// lettercase percentage ratio in c for code eval by steven a dunn

#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(int argc, char* argv[])
{
    FILE *fp = fopen(argv[1], "r");
    if (fp != NULL)
    {
        char line[1024];
        while (fgets(line, sizeof(line), fp))
        {
            int upper_count = 0;
            int lower_count = 0;

            for (int i = 0; i < strlen(line); ++i)
            {
                char c = line[i];
                if (isupper(c))
                    upper_count++;
                else if (islower(c))
                    lower_count++;
            }

            int total_count = upper_count + lower_count;
            float upper_percent = ((float)upper_count / total_count) * 100.0;
            float lower_percent = ((float)lower_count / total_count) * 100.0;

            printf("lowercase: %.2f uppercase: %.2f \n", lower_percent, upper_percent);
        }
        fclose(fp);
    }
    return 0;
}